language-server/src/main/java/org/sonarlint/languageserver/SonarLintLanguageServer.java

```diff
@@ -286,6 +296,103 @@ private static TextDocumentSyncOptions getTextDocumentSyncOptions() {
     return textDocumentSyncOptions;
   }
 
+  private void handleUpdateServerStorageCommand(@Nullable List<Object> arguments) {
+    engineCache.clearConnectedEngines();
+
+    serverInfoCache.replace(arguments);
+
+    serverInfoCache.forEach((serverId, serverInfo) -> {
+      ConnectedSonarLintEngine engine = engineCache.getOrCreateConnectedEngine(serverInfo);
+      if (engine == null) {
+        popupError("Could not start server: " + serverId);
+      } else {
+        updateServerStorage(engine, serverInfo);
+      }
+    });
+  }
+
+  private void updateServerStorage(ConnectedSonarLintEngine engine, ServerInfo serverInfo) {
+    String serverId = serverInfo.serverId;
+    debug("Updating global storage of server " + serverId + ", may take some time...");
+    ServerConfiguration serverConfig = getServerConfiguration(serverInfo);
+    engine.update(serverConfig, null);
+    debug("Successfully updated global storage of server " + serverId);
+  }
+
+  private ServerConfiguration getServerConfiguration(ServerInfo serverInfo) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Make "getServerConfiguration" a "static" method. ([squid:S2325](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2325))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarlint.core%3Asonarlint-core-parent&pullRequest=192&issues=AWND8neO6NykrOWnW1BW&open=AWND8neO6NykrOWnW1BW)

[ISSUE_KEY]: AWND8neO6NykrOWnW1BW